src/main/java/org/threadly/db/aurora/AuroraDataSource.java

```diff
@@ -0,0 +1,321 @@
+package org.threadly.db.aurora;
+
+import java.io.PrintWriter;
+import java.io.Serializable;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import java.util.Properties;
+import java.util.logging.Logger;
+
+import javax.naming.NamingException;
+import javax.naming.Reference;
+import javax.naming.Referenceable;
+import javax.naming.StringRefAddr;
+import javax.sql.DataSource;
+
+public final class AuroraDataSource implements DataSource, Referenceable, Serializable {
+  
+  private static final String AURORA_PREFIX = "jdbc:mysql:aurora:";
+  
+  private static final long serialVersionUID = 1L;
+  
+  /** The driver to create connections with */
+  private static final NonRegisteringDriver driver;
```

Comment:
Why do we want to avoid registering the driver with the `DriverManager`?  I see this is what the mysql driver is doing as well, but no documentation I see that explains why.  Thoughts?

Reply:
There is no need to register the driver when using application server JNDI, so it is better to use this NonRegistering driver.